src/Service/DeliveryManager.php

```diff
@@ -146,4 +135,79 @@ public function setDefaults(Delivery $delivery)
 
         $delivery->setDistance(ceil($distance));
     }
+
+    public function createTasksFromRecurrenceRule(Task\RecurrenceRule $recurrenceRule, string $startDate, bool $persist = true): array
+    {
+        $store = $recurrenceRule->getStore();
+
+        $template = $recurrenceRule->getTemplate();
+        $tasksTemplates = $template['@type'] === 'hydra:Collection' ?
+            $template['hydra:member'] : [ $template ];
+
+        $tasksTemplates = array_map(function ($taskTemplate) use ($startDate) {
+
+            $taskTemplate['after'] = (new \DateTime($startDate . ' ' . $taskTemplate['after']))->format(\DateTime::ATOM);
+            $taskTemplate['before'] = (new \DateTime($startDate . ' ' . $taskTemplate['before']))->format(\DateTime::ATOM);
+
+            return $taskTemplate;
+        }, $tasksTemplates);
+
+
+        $tasks = [];
+        foreach ($tasksTemplates as $payload) {
+
+            $task = $this->denormalizer->denormalize($payload, Task::class, 'jsonld', [
+                'groups' => ['task_create']
+            ]);
+
+            $task->setOrganization($store->getOrganization());
+            $task->setRecurrenceRule($recurrenceRule);
+
+            if ($persist) {
+                $this->entityManager->persist($task);
+            }
+
+            $tasks[] = $task;
+        }
+
+        return $tasks;
+    }
+
+    public function createDeliveryFromRecurrenceRule(Task\RecurrenceRule $recurrenceRule, string $startDate, bool $persist = true): ?Delivery
+    {
+        $store = $recurrenceRule->getStore();
+        $tasks = $this->createTasksFromRecurrenceRule($recurrenceRule, $startDate, $persist);
+
+        $delivery = null;
+        if (count($tasks) > 1 && $tasks[0]->isPickup()) {
+            $delivery = Delivery::createWithTasks(...$tasks);
+            $store->addDelivery($delivery);
+
+            $this->calculateRoute($delivery);
+
+            if ($persist) {
+                $this->entityManager->persist($delivery);
+            }
+        }
+
+        return $delivery;
+    }
+
+    public function calculateRoute(TaskCollectionInterface $taskCollection): void
```

Comment:
wouldnt that be calculated automatically by `TaskListSubscriber` ?

Reply:
`TaskListSubscriber` is triggered only when an Entity is persisted. On the recurrence rules page, I only create  placeholders for Orders/Deliveries/Tasks, but don't persist them